cips/cip-start-times.md

```diff
@@ -0,0 +1,63 @@
+
+## Specification
+
+The following modifications are proposed:
+
+- Validators select a vote time that is their `cs.StartTime` as opposed to their current time when they vote for a block. `cs.StartTime` is the time that a validator would propose a block if they were the proposer and the time with which `TimeoutPropose` starts.
+  - In the case, that `cs.StartTime` is after the locked or proposal block time, the validator will pick a time that is 1 millisecond later to satisfy the property of increasing block times.
+  - In the case of the first block, the start time will be the current time.
+- Upon receiving 2f + 1 PRECOMMIT votes, each validator calculates the **local** stake-weighted median start time from those votes. It sets the new start time of the next height to be the aforementioned calculated time plus the `TargetInterval`.
+  - If the nodes local time is after the new start time, then the node will start immediately. This will be the case for heights that require multiple rounds to come to consensus.
+- A proposer MUST include a block time in the header that is equal to the stake-weighted median start time of the precommit votes that are included in the LastCommit of the proposed block + the `TargetInterval`.
+  - All validators MUST verify that the block time has been calculated as per the above rules. If not, the block will not be accepted.
+- `TargetInterval` is part of `BlockParams` within `ConsensusParams`. A `TargetInterval` of zero signifies that this mechanism is disabled. The `TargetInterval` CAN be set in a hard fork upgrade when all nodes are running the latest version, activating the new timing mechnanism.
+- The `TargetInterval` once activated will be set to 12 seconds. `TargetInterval` can not be modified via governance and SHOULD only change in major version upgrades.
+- `TimeoutCommit` will be ignored when this mechanism is active.
+
+To help develop an intuition on what this proposal changes, if we were to plot all the start times of each validator in the network in order, assuming equal weighting, the start times of all nodes for the **next height** will be within the "middle third of start times" of each other. This has a converging effect even in cases where consensus takes longer than the `TargetInterval` itself.
+
+## Parameters
+
+| Parameter     | Proposed value | Description                                                                                                                | Changeable via Governance |
+|---------------|---------|------------------------------------------------------------------------------------------------------------------------|---------------------------|
+| BlockParams.TargetIntervalMS | 12000 milliseconds (12 seconds)  | The target interval between start times in milliseconds                                                            | false                     |
```

Comment:
would it make sense to mention that we're confident that this mechanism will allow 12 seconds for blocks <=8mb or whatever the upper bound is?

The upper bound can be defined as the biggest block the network can process in 12 sec if they start processing it right away.